module/VuFindSearch/src/VuFindSearch/Backend/Blender/Response/Json/RecordCollectionFactory.php

```diff
@@ -0,0 +1,106 @@
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   David Maus <maus@hab.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+namespace VuFindSearch\Backend\Solr\Response\Json;
+
+use VuFindSearch\Exception\InvalidArgumentException;
+use VuFindSearch\Response\RecordCollectionFactoryInterface;
+
+/**
+ * Simple JSON-based factory for record collection.
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   David Maus <maus@hab.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+class RecordCollectionFactory implements RecordCollectionFactoryInterface
+{
+    /**
+     * Factory to turn data into a record object.
+     *
+     * @var Callable
+     */
+    protected $recordFactory;
+
+    /**
+     * Class of collection.
+     *
+     * @var string
+     */
+    protected $collectionClass;
+
+    /**
+     * Constructor.
+     *
+     * @param Callable $recordFactory   Callback to construct records
+     * @param string   $collectionClass Class of collection
+     *
+     * @return void
+     */
+    public function __construct($recordFactory = null, $collectionClass = null)
+    {
+        if (null === $recordFactory) {
+            $this->recordFactory = function ($data) {
+                return new Record($data);
+            };
+        } else {
+            $this->recordFactory = $recordFactory;
+        }
+        $this->collectionClass = null === $collectionClass
+            ? 'VuFindSearch\Backend\Blender\Response\Json\RecordCollection'
+            : $collectionClass;
+    }
+
+    /**
+     * Return record collection.
+     *
+     * @param array $response Deserialized JSON response
+     *
+     * @return RecordCollection
+     */
+    public function factory($response)
+    {
+        if (!is_array($response)) {
+            throw new InvalidArgumentException(
+                sprintf(
+                    'Unexpected type of value: Expected array, got %s',
+                    gettype($response)
+                )
+            );
+        }
+        $collection = new $this->collectionClass($response);
+        if (isset($response['response']['docs'])) {
+            foreach ($response['response']['docs'] as $doc) {
```

Comment:
`foreach ($response['response']['docs'] ?? [] as $doc) {` could eliminate the isset check.

Reply:
Changed. This comes from Solr's factory.

Reply:
Original factory updated in #2348.